app/helpers/projects_helper.rb

```diff
@@ -3,4 +3,23 @@ def estimated(story)
     @estimate_id = story.estimate_for(current_user)
     @estimate_id.present?
   end
+
+  def link_unless_archived(project, text, url, classes: nil, method: :get, remote: false, icon: nil, data_attr: nil, id: nil)
+    link_text = text_content(icon, text)
+    if project.archived? || project.parent&.archived?
+      button_tag(link_text, disabled: true, class: "button #{classes} disabled", aria: {disabled: true})
+    else
+      link_to(link_text, url, class: "button #{classes}", title: text, method: method, remote: remote, data: data_attr, id: id)
+    end
+  end
```

Comment:
This may be a personal way to do things in this scenario, so, maybe just for the record.

Pros:
  - you can pass more data if desired.
Cons:
  - many changes, again (and we may don't want that).
  - need to test if what I put as example works as expected.

```suggestion
  def link_unless_archived(project, text, url, clases: nil, options = { method: :get, remote: false })
    link_text = text_content(icon, text)
    if project.archived? || project.parent&.archived?
      button_tag(link_text, { disabled: true, aria: disabled: true, class: "button #{classes} disabled" }.merge(options))
    else
      link_to(link_text, url, { class: "button #{classes}" }.merge(options))
    end
  end
```

Reply:
I like this suggestion @JuanVqz, I do think it is a lot of changes - so not worth our time at this stage?